CustomDeserializers/AvroExample/EventhubCaptureCustomEvent.cs

```diff
@@ -0,0 +1,53 @@
+﻿//********************************************************* 
+// 
+//    Copyright (c) Microsoft. All rights reserved. 
+//    This code is licensed under the Microsoft Public License. 
+//    THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF 
+//    ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY 
+//    IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR 
+//    PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT. 
+// 
+//********************************************************* 
+
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+
+using ExampleCustomCode.Serialization;
+
+namespace ExampleCustomCode.AvroSerialization
+{
+    // Reads eventhub capture in avro format and deserializes EventData.Body as a simple record.
+    public sealed class EventhubCaptureCustomEvent : EventhubCaptureReader<EventHubRecord>
```

Comment:
EventhubCaptureCustomEventReader

Reply:
changed.

Reply:
I added specific checks and errors in next iteration. 
Even without these explicit diagnostics, failures from custom deserializers would be captured. However I think adding the diagnostics at specific places 1) Shows the ability and 2) Provides better context - one error is likely due to invalid path, other error is likely due to a format change that breaks older readers.